common/src/main/java/com/wynntils/models/seaskipper/SeaskipperModel.java

```diff
@@ -0,0 +1,202 @@
+
+    private static AbstractContainerScreen actualSeaskipperScreen;
+
+    private final Map<SeaskipperDestinationItem, Integer> destinations = new HashMap<>();
+    private final List<SeaskipperPoi> seaskipperPois = new ArrayList<>();
+    private final List<String> availableNames = new ArrayList<>();
+
+    private List<SeaskipperPoi> availableDestinations = new ArrayList<>();
+    private SeaskipperPoi currentPoi;
+    private int boatSlot;
+    private int containerId = -2;
+
+    public SeaskipperModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
+    }
+
+    @SubscribeEvent
+    public void onScreenOpened(ScreenOpenedEvent.Post event) {
+        if (Models.Container.isSeaskipper(
+                ComponentUtils.getUnformatted(event.getScreen().getTitle()))) {
+            actualSeaskipperScreen = (AbstractContainerScreen) event.getScreen();
+
+            loadSeaskipperPois();
+        }
+    }
+
+    @SubscribeEvent
+    public void onMenuOpened(MenuEvent.MenuOpenedEvent event) {
+        if (Models.Container.isSeaskipper(ComponentUtils.getUnformatted(event.getTitle()))) {
+            containerId = event.getContainerId();
+        }
+    }
+
+    public void closeScreen() {
+        ContainerUtils.closeContainer(actualSeaskipperScreen.getMenu().containerId);
+    }
+
+    @SubscribeEvent
+    public void onSetSlot(ContainerSetSlotEvent event) {
+        if (event.getContainerId() != containerId) return;
+
+        if (StyledText.fromComponent(event.getItemStack().getHoverName()).equals(OAK_BOAT_NAME)) {
+            setBoatSlot(event.getSlot());
+            return;
+        }
+
+        Matcher matcher =
+                StyledText.fromComponent(event.getItemStack().getHoverName()).getMatcher(SEASKIPPER_PASS_PATTERN);
+        if (!matcher.matches()) return;
+
+        String destination = matcher.group(1);
+        int price = Integer.parseInt(matcher.group(2));
+        String shorthand = destination.substring(0, 2);
+
+        addDestination(new SeaskipperDestinationItem(destination, price, shorthand), event.getSlot());
+
+        availableNames.add(destination);
+    }
+
+    private void addDestination(SeaskipperDestinationItem destination, int slot) {
+        destinations.put(destination, slot);
+    }
+
+    public Map<SeaskipperDestinationItem, Integer> getDestinations() {
+        return Collections.unmodifiableMap(destinations);
+    }
+
+    public List<SeaskipperPoi> getAvailableDestinations() {
+        return Collections.unmodifiableList(availableDestinations);
+    }
+
+    public List<SeaskipperPoi> getSeaskipperPois() {
+        return Collections.unmodifiableList(seaskipperPois);
+    }
+
+    private void setBoatSlot(int boatSlot) {
+        this.boatSlot = boatSlot;
+    }
+
+    public void purchaseBoat() {
+        clickSlot(boatSlot);
+    }
+
+    public SeaskipperPoi getCurrentPoi() {
+        return currentPoi;
+    }
+
+    public void buyPass(String destinationToTravelTo) {
```

Comment:
```suggestion
    public void purchasePass(String destinationToTravelTo) {
```

... or rename the other method "buyBoat".